### PR TITLE
[Buildkite] Add wheels, jars, and docker builds.

### PR DESCRIPTION
**Pull request by simon-mo**
This PR adds new wheels, jars, and docker builds. It will build all three but only push wheels/jars to designated S3 repo on branch build. (we are not pushing docker yet because we don't want to override Travis pushes)

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+    - if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
+    - pip install -q docker aws_requests_auth boto3
+    - python .buildkite/copy_files.py --destination wheels --path ./.whl
+    - python .buildkite/copy_files.py --destination wheels --path ./.jar/linux
+
+- label: ":docker: Build Images"
+  conditions: ["RAY_CI_LINUX_WHEELS_AFFECTED"]
+  commands:
+    - LINUX_WHEELS=1 LINUX_JARS=1 DOCKER_BUILD_PY37=1 RAY_INSTALL_JAVA=1 ./ci/travis/ci.sh build
+    - python ./ci/travis/build-docker-images.py PY37
```
Comment (ijrsvt) [inline .buildkite/pipeline.yml:23]: Can we also build PY36_PY38?

File: ci/travis/build-docker-images.py [hunk 2]
```diff
@@ -32,11 +32,14 @@
...
 def _merge_build():
+    if os.environ.get("BUILDKITE"):
+        return os.environ.get("BUILDKITE_PULL_REQUEST").lower() == "false"
     return os.environ.get("TRAVIS_PULL_REQUEST").lower() == "false"
 
 
 def _release_build():
-    branch = os.environ.get("TRAVIS_BRANCH")
+    branch = os.environ.get("TRAVIS_BRANCH",
+                            os.environ.get("BUILDKITE_BRANCH"))
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:42]: Can we do something like 
```
os.environ.get("TRAVIS_BRANCH") or os.environ.get("BUILDKITE_BRANCH")
```

File: ci/travis/build-docker-images.py [hunk 3]
```diff
@@ -32,11 +32,14 @@
 
 
 def _merge_build():
+    if os.environ.get("BUILDKITE"):
+        return os.environ.get("BUILDKITE_PULL_REQUEST").lower() == "false"
     return os.environ.get("TRAVIS_PULL_REQUEST").lower() == "false"
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:37]: Not necessary for this PR, but I honestly think we should switch to passing args to build-docker-images, **not** reading a bunch of env variables

**Review by ijrsvt — COMMENTED** [review]
A few quick comments!

File: python/ray/serve/utils.py [hunk 4]
```diff
@@ -378,7 +378,10 @@ def import_attr(full_path: str):
 
 
 async def mock_imported_function(batch):
-    return [await request.body() for request in batch]
+    result = []
```
Comment (simon-mo) [inline python/ray/serve/utils.py:381]: lint

**Comment by simon-mo** [comment]
@ijrsvt thanks! addressed.

File: python/ray/serve/utils.py (hunk 4 above)
Comment (ijrsvt) [reply python/ray/serve/utils.py:381]: OOC why does this not work on BuildKite?

**Review by ijrsvt — APPROVED** [review]

**Pull request merged**